tests/pipelines/main_gm_threshold.nf.test

```diff
@@ -0,0 +1,135 @@
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet1.csv"
+                outdir = "results"
+
+                gm_thresholds = null
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stdout.contains("ERROR ~ --gm_thresholds null: Cannot pass null or empty string")
+        }
+    }
+
+    test("Test fail pipeline if empty threshold set") {
+        tag "pipeline_failure_no_threshold"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet1.csv"
+                outdir = "results"
+
+                gm_thresholds = ""
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stdout.contains("ERROR ~ --gm_thresholds : Cannot pass null or empty string")
+        }
+    }
+
+    test("Test fail pipeline if negative threshold set") {
+        tag "pipeline_failure_negative_threshold"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet1.csv"
+                outdir = "results"
+
+                gm_thresholds = "-1"
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stderr.contains('* --gm_thresholds: string [-1] does not match pattern ^(\\d+(\\.\\d+)?,)*\\d+(\\.\\d+)?$ (-1)')
+        }
+    }
+
+    test("Test fail pipeline if mismatch between thresholds and scaled distm") {
+        tag "pipeline_failure_threshold_scaled"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet1.csv"
+                outdir = "results"
+
+                gm_thresholds = "1,0.5,2"
+                pd_distm = "scaled"
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stdout.contains("ERROR ~ '--pd_distm scaled' is set, but '--gm_thresholds 1,0.5,2' contains thresholds outside of range [0,1]."
+                                            + " Please either set '--pd_distm hamming' or adjust the threshold values.")
+        }
+    }
+
+    test("Test fail pipeline if mismatch between thresholds and hamming distm") {
+        tag "pipeline_failure_threshold_hamming"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet1.csv"
+                outdir = "results"
+
+                gm_thresholds = "2,1,0.5"
+                pd_distm = "hamming"
+            }
+        }
+
+        then {
+            assert workflow.failed
+            assert workflow.stdout.contains("ERROR ~ '--pd_distm hamming' is set, but '--gm_thresholds 2,1,0.5' contains fractions."
+                                            + " Please either set '--pd_distm scaled' or remove fractions from distance thresholds.")
+        }
+    }
+
+    test("Test pipeline with single threshold set to 1") {
```

Comment:
Could you change the title of this test to indicate you are testing a failure condition? Could you also test an equivalent success condition (as in a case where the supplied threshold does equal the number of threshold columns)? Or is this tested elsewhere?

Reply:
Done here: [c57ef6c](https://github.com/phac-nml/gasnomenclature/pull/15/commits/c57ef6c0f1a1817ec11e7782369a133e20e8adfd)